Display slice plane

mathturtle opened the pull request:
This PR is work to address #92.  It displays a point on the plane and the normal vector of the non-orthogonal slice in the module properties panel.  @Hovden how is this?

Hovden commented:
This is great and very useful. Several researchers at Cornell are using tomviz for analysis daily and non-ortho slicing is popular.

Some minor comments:
1) When changing the point on plane, the slice is shifted and the render is not properly refreshed until the slice it is moved.
2) It is not clear which is the x,y,z coordinate.
3) There are a lot of significant digits displayed.

To solve 2&3, it may be worth breaking the everything into individual rows and labeling "x", "y", "z" to the left of each input box. The added space will accommodate larger boxes and better address point 3.

Hovden commented:
It would be sufficient to simply label x, y, z across the top and merge.

mathturtle commented:
@Hovden 

1) Good catch, I'll dig into this one.
2/3) This is going to require changing ParaView to support labeling the components of this kind of property.  It will actually be easier to put "x", "y" and "z" to the left of each input box than across the top.

mathturtle commented:
Ok, pushed a new version that fixes the update issue.  I was using the wrong function to update the plane.

Do you want to wait on the ParaView changes to add the coordinate labels before merging this?

Hovden commented:
I'm happy to merge if the code looks good. +1

mathturtle commented:
I have the ParaView changes working [here](https://gitlab.kitware.com/paraview/paraview/merge_requests/379).  They may be changed some in code review though.  Let's wait until they are finalized and I can add a 'use component labels' flag to those properties.  I suspect @utkarshayachit may want me to change how that flag is formatted.  Here is a preview:
![component_labels](https://cloud.githubusercontent.com/assets/553788/9885897/a11981f8-5bb5-11e5-9ab6-ccb65b65480a.png)

mathturtle commented:
We updated the ParaView code a bit so that the labels are aligned vertically with the property label on the left.  It still looks bad, but at least it is consistent.  I've added the commit here that will cause labels to appear when ParaView is updated.

New preview :(
![component_labels2](https://cloud.githubusercontent.com/assets/553788/9908320/1fd066aa-5c61-11e5-90b5-16b9685b4558.png)

mathturtle commented:
The ParaView fixes are merged and [this superbuild branch](https://github.com/OpenChemistry/tomviz-superbuild/pull/51) adds the fix.

cryos commented on tomviz/ModuleSlice.cxx:
```diff
@@ -48,7 +48,9 @@ namespace tomviz
-ModuleSlice::ModuleSlice(QObject* parentObject) : Superclass(parentObject)
+ModuleSlice::ModuleSlice(QObject* parentObject)
+  : Superclass(parentObject)
+  , IgnoreSignals(false)
```
This looks odd, I think the comma should always be at the end of the line, not the start of the following.

cryos commented on tomviz/ModuleSlice.cxx:
```diff
@@ -318,6 +324,12 @@ bool ModuleSlice::deserialize(const pugi::xml_node& ns)
 {
+  // Avoid recursive clobbering of the plane position
+  if (this->IgnoreSignals)
+    {
```
Indentation

cryos commented:
Looks good to me, tiny niggle on comma, fix in the follow up.
